salt/utils/process.py

```diff
@@ -55,7 +56,17 @@ def notify_systemd():
         import systemd.daemon
     except ImportError:
         if salt.utils.which('systemd-notify') and systemd_notify_call('--booted'):
-            return systemd_notify_call('--ready')
+            # Notify systemd synchronously
+            notify_socket = os.getenv('NOTIFY_SOCKET')
+            if notify_socket:
+                # Handle abstract namespace socket
+                if notify_socket.startswith('@'):
+                    notify_socket = '\0{0}'.format(notify_socket[1:])
+                sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+                sock.connect(notify_socket)
+                sock.sendall('READY=1'.encode())
+                sock.close()
+                return True
```

Comment:
I think it makes sense to fallback on `systemd_notify_call('--ready')` when `socket` raises `socket.error` for any reason.